### DISCUSSION TITLE
感觉 validate 中不应该使用表单的 pattern 做逻辑处理

### DISCUSSION THREAD

**Discussion by liuweiGL** [alibaba/formily#3776] [Q&A 2.x]
## 问题
form.validate 时会判断表单是否为 "editable" 状态，导致某些场景下功能受限：
https://github.com/alibaba/formily/blob/1f33ebc55e1074c678bf7bdd8bceee315c704c34/packages/core/src/shared/internals.ts#L889

## 举例
在详情模式中允许使用 [Editable](https://antd.formilyjs.org/zh-CN/components/editable) 对某个字段变更

## 向后兼容
 提供一个  `validatePattern` 配置项，使用方式类似 `validateFirst` 。并为其设置特殊值为 “*”  默认值为 “editable”， 是否可行？

**Comment by janryWang** [comment]
可以，这个问题挺多用户提了，增加一个选项即可，麻烦来个 PR ？

**Comment by liuweiGL** [comment]
恩，争取最近抽时间，今年有点忙😂

**Comment by liuweiGL** [comment]
https://github.com/alibaba/formily/pull/4211
